cpp/include/proxy_sequences.h

```diff
@@ -0,0 +1,133 @@
+/*
+  Copyright (C) 2016 iNuron NV
+
+  This file is part of Open vStorage Open Source Edition (OSE), as available
+  from
+
+
+  http://www.openvstorage.org and
+  http://www.openvstorage.com.
+
+  This file is free software; you can redistribute it and/or modify it
+  under the terms of the GNU Affero General Public License v3 (GNU AGPLv3)
+  as published by the Free Software Foundation, in version 3 as it comes
+  in the <LICENSE.txt> file of the Open vStorage OSE distribution.
+
+  Open vStorage is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY of any kind.
+*/
+
+#pragma once
+
+#include "checksum.h"
+#include "llio.h"
+#include <memory>
+
+namespace alba {
+namespace proxy_client {
+namespace sequences {
+
+class Assert {
+public:
+  virtual ~Assert(){};
+  virtual void to(llio::message_builder &) const = 0;
+};
+
+class AssertObjectExists : public Assert {
+public:
+  AssertObjectExists(std::string name) : _name(name){};
+
+  void to(llio::message_builder &mb) const {
+    mb.add_type(1);
+    llio::to(mb, _name);
+  }
+
+  std::string _name;
+};
+
+class AssertObjectDoesNotExist : public Assert {
+public:
+  AssertObjectDoesNotExist(std::string name) : _name(name){};
+
+  void to(llio::message_builder &mb) const {
+    mb.add_type(2);
```

Comment:
NB: not an issue, but you might want to consider using C++-11's `override` and `final` specifiers for virtual methods to give the compiler extra hints.